### PR TITLE
feat(shim): forward ro bind-mount option to virtio-fs share

### PR DESCRIPTION
**Pull request by ndeloof**
## Summary

- Parse the `ro` OCI mount option in `bindMounter.FromBundle` and carry it on `bindMount`, mirroring what the ext4 branch in the same file already does.
- Forward the flag through `sandbox.WithFS` to `vmInstance.AddFS`, which now applies `vm.MountOpts` before calling into libkrun.
- Bind the optional `krun_add_virtiofs3` symbol (libkrun >= 1.18) via a small `registerOptionalLibFunc` helper and a `,optional` struct-tag flag; route `AddVirtiofs` through it when present so the share is marked read-only at the host edge.
- Fall back to `krun_add_virtiofs` with a single warn log when the v3 symbol is not exported; guest-side `ro` enforcement via OCI mount options is unchanged.
- Cover the new behavior with unit tests for `parseCTag` and the `AddVirtiofs` v3-vs-legacy matrix (prefers v3, falls back when v3 is nil, surfaces non-zero return codes on both paths, errors when neither symbol is bound).

## Behavior

| libkrun version | Effect of this change |
| --- | --- |
| `< 1.18` (no `krun_add_virtiofs3` exported, e.g. the v1.17.4 currently pinned) | No-op on the host side. The fallback path runs and logs a single warn for read-only shares; the guest continues to enforce `ro` through the OCI mount options exactly as today. |
| `>= 1.18` (`krun_add_virtiofs3` exported) | The virtio-fs share is marked read-only at the host edge, so the VMM rejects writes in addition to the guest bind mount. This is defense-in-depth and lets the VMM treat the share as immutable for any future caching optimizations. |

## Test plan

- [ ] `go build ./...`
- [ ] `go test ./internal/shim/task/... ./internal/vm/libkrun/...`

### CODE REVIEW

File: internal/vm/libkrun/krun_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+		{"", "", false},
+	}
+	for _, c := range cases {
+		t.Run(c.in, func(t *testing.T) {
+			name, opt := parseCTag(c.in)
+			if name != c.wantName || opt != c.wantOpt {
+				t.Fatalf("parseCTag(%q) = (%q, %v), want (%q, %v)", c.in, name, opt, c.wantName, c.wantOpt)
+			}
+		})
+	}
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply internal/vm/libkrun/krun_test.go:44]: Fixed in cb20f8b — switched the table to named subtests (`empty`, `plain`, etc.) so `t.Run` no longer receives an empty string for the `""` case.

File: internal/shim/task/mount.go [hunk 2]
```diff
@@ -242,10 +243,24 @@ func (bm *bindMounter) FromBundle(ctx context.Context, b *bundle.Bundle) error {
...
+		// applies the `ro` mount option to enforce read-only inside the
+		// container; this flag is defense in depth and ensures the host
+		// also rejects writes when the underlying libkrun supports it.
+		readOnly := false
+		for _, opt := range m.Options {
+			if opt == "ro" {
+				readOnly = true
+				break
+			}
+		}
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply internal/shim/task/mount.go:262]: Fixed in cb20f8b — the loop now scans every option and lets later `ro`/`rw` override earlier ones (no `break`), matching standard mount-option semantics. Added two regression tests (`bind mount ro then rw` and `bind mount rw then ro`) in `mount_test.go` to lock the behavior down.

**Comment by ndeloof** [comment]
Downgraded the fallback log from warn to info: missing krun_add_virtiofs3 isn't a blocker — guest-side enforcement still applies. New head: 2f60004.

**Comment by ndeloof** [comment]
Branch was renamed to drop the `nicolasdeloof/` prefix, which auto-closed this PR (GitHub does not propagate fork-branch renames to cross-repo PRs). Recreated as #205 with the same commit (2f60004f5e2e064c4ff72c0e293e25dac0669cc3) on branch `ndeloof:virtiofs-ro-host-edge`. Please continue review there.
